Build failure: wezterm

### Steps To Reproduce

Steps to reproduce the behavior:
1. build wezterm with today's master branch (I'll be able to find the exact commit later)

### Build log

```
building '/nix/store/l4ha5w08li3dj3s5jf0qr9rl7pd1f3pn-wezterm-20240203-110809-5046fc22.drv'...
note: keeping build directory '/tmp/nix-build-wezterm-20240203-110809-5046fc22.drv-16/build'
error: builder for '/nix/store/l4ha5w08li3dj3s5jf0qr9rl7pd1f3pn-wezterm-20240203-110809-5046fc22.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling native-tls v0.2.11
       >    Compiling line-wrap v0.1.1
       >    Compiling cairo-sys-rs v0.18.0 (/build/source/deps/cairo)
       >    Compiling x11 v2.21.0
       >    Compiling quick-xml v0.31.0
       >    Compiling quick-xml v0.30.0
       >    Compiling wayland-commons v0.29.5
       > For more information about this error, try `rustc --explain E0282`.
       > error: could not compile `time` (lib) due to 1 previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/l4ha5w08li3dj3s5jf0qr9rl7pd1f3pn-wezterm-20240203-110809-5046fc22.drv'.
```

```
error[E0282]: type annotations needed for `Box<_>`
  --> /build/cargo-vendor-dir/time-0.3.31/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling toml_edit v0.19.15
   Compiling memoffset v0.9.0
   Compiling fixed v1.24.0
   Compiling rustc-demangle v0.1.23
   Compiling safemem v0.3.3
   Compiling scoped-tls v1.0.1
   Compiling native-tls v0.2.11
   Compiling line-wrap v0.1.1
   Compiling cairo-sys-rs v0.18.0 (/build/source/deps/cairo)
   Compiling x11 v2.21.0
   Compiling quick-xml v0.31.0
   Compiling quick-xml v0.30.0
   Compiling wayland-commons v0.29.5
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

@SuperSandro2000 @mimame

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.4, NixOS, 24.11 (Vicuna), 24.11.20240814.29aa9e9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `""`
 - nixpkgs: `/nix/store/5d90zrvwna6sxi32sqvdhi0d665kka44-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/terminal-emulators/wezterm/default.nix
+++ b/pkgs/applications/terminal-emulators/wezterm/default.nix
@@ -43,6 +43,8 @@ rustPlatform.buildRustPackage rec {
   };
 
   postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+
     echo ${version} > .tag
 
     # tests are failing with: Unable to exchange encryption keys

